### PR TITLE
Add v-prediction to Flax training examples

### PR DESCRIPTION
**Pull request by infamix**
Added v-prediction to FlaxDDIM and FlaxDDPM schedulers as well as the Flax training examples.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Gently ping @pcuenca and @yiyixuxu here

File: src/diffusers/schedulers/scheduling_ddim_flax.py [hunk 1]
```diff
@@ -322,5 +322,22 @@ def add_noise(
...
+        sqrt_alpha_prod = self.alphas_cumprod[timesteps] ** 0.5
+        sqrt_alpha_prod = sqrt_alpha_prod.flatten()
+        sqrt_alpha_prod = broadcast_to_shape_from_left(sqrt_alpha_prod, sample.shape)
+
+        sqrt_one_minus_alpha_prod = (1 - self.alphas_cumprod[timesteps]) ** 0.5
+        sqrt_one_minus_alpha_prod = sqrt_one_minus_alpha_prod.flatten()
+        sqrt_one_minus_alpha_prod = broadcast_to_shape_from_left(sqrt_one_minus_alpha_prod, sample.shape)
+
+        velocity = sqrt_alpha_prod * noise - sqrt_one_minus_alpha_prod * sample
+        return velocity
```
Comment (patil-suraj) [inline src/diffusers/schedulers/scheduling_ddim_flax.py:340]: Looks good, but we'll need to refactor this into a common function so like `add_noise_common` cf #1661

We should add a `get_velocity_common` function to `scheduling_flax_util.py` like `add_noise_common` https://github.com/huggingface/diffusers/blob/main/src/diffusers/schedulers/scheduling_utils_flax.py#L245 and then call that function from `get_velocity`.


after this, in the training script, we'll need to call this method using
```python
noise_scheduler.get_velocity(noise_scheduler_state, latents, noise, timesteps)
```

File: src/diffusers/schedulers/scheduling_ddpm_flax.py [hunk 2]
```diff
@@ -314,5 +314,22 @@ def add_noise(
...
+    ) -> jnp.ndarray:
+        sqrt_alpha_prod = self.alphas_cumprod[timesteps] ** 0.5
+        sqrt_alpha_prod = sqrt_alpha_prod.flatten()
+        sqrt_alpha_prod = broadcast_to_shape_from_left(sqrt_alpha_prod, sample.shape)
+
+        sqrt_one_minus_alpha_prod = (1 - self.alphas_cumprod[timesteps]) ** 0.5
+        sqrt_one_minus_alpha_prod = sqrt_one_minus_alpha_prod.flatten()
+        sqrt_one_minus_alpha_prod = broadcast_to_shape_from_left(sqrt_one_minus_alpha_prod, sample.shape)
+
+        velocity = sqrt_alpha_prod * noise - sqrt_one_minus_alpha_prod * sample
```
Comment (patil-suraj) [inline src/diffusers/schedulers/scheduling_ddpm_flax.py:331]: same comment as above.

**Review by patil-suraj — CHANGES_REQUESTED** [review]
Thanks a lot for working on this, looks good. Just left one comment, we'll need to add a common function to get the velocity after the changes in #1661 , after that this should be good to merge.

**Comment by patil-suraj** [comment]
We'll also need to resolve the conflict and rebase to `main`. Let me know if you need any help.

File: src/diffusers/schedulers/scheduling_ddim_flax.py (hunk 1 above)
Comment (infamix) [reply src/diffusers/schedulers/scheduling_ddim_flax.py:340]: I'd suggest merging PR [#1783](https://github.com/huggingface/diffusers/pull/1783) first, the changes there are clearly the correct ones.

Comment (patil-suraj) [reply src/diffusers/schedulers/scheduling_ddim_flax.py:340]: cool, this PR was opened first so I preferred this, but if you are fine with it, we'll go ahead with #1783

Comment (infamix) [reply src/diffusers/schedulers/scheduling_ddim_flax.py:340]: I am fine with it, go ahead.

**Pull request closed without merging**